src/test/frontend/deploy/createsecret_controller_test.js

```diff
@@ -0,0 +1,125 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import CreateSecretController from 'deploy/createsecret_controller';
+import DeployLabel from 'deploy/deploylabel';
+import deployModule from 'deploy/deploy_module';
+
+describe('Create-Secret dialog', () => {
+
+  let ctrl;
+  let httpBackend;
+  beforeEach(() => {
+    angular.mock.module(deployModule.name);
+    angular.mock.inject(($controller, $httpBackend) => {
+      ctrl = $controller(CreateSecretController, {'namespace': 'default'});
+      httpBackend = $httpBackend;
+    });
+  });
+
+  /**
+   * Test if there is always at least one available key-value input field.
+   */
+  it('there should be at least one key-value pair input',
+     () => { expect(ctrl.labels.length > 0).toBeTruthy(); });
+
+  /**
+   * Test if the submit button of the create form is disabled properly.
+   * A secret should not be created before both the secret name and data
+   * have been given by the user. The key-value pairs of the data must be
+   * non empty (concerning both key and value).
+   */
+  it('should disable ok-button if secret name and data are missing', () => {
+    // given an empty secret name
+    ctrl.secretName = '';
+    // a couple of non-complete key-value pairs (no data)
+    ctrl.labels = [
+      new DeployLabel("foo", ""),
+      new DeployLabel("", "bar"),
+    ];
+    let result = ctrl.isDisabled();
+    expect(result).toBeTruthy();
```

Comment:
Better test with `toBe(true)`. This makes the tests more strict. Truthly values are `"  "`, `{}`, etc.